cli/command/stack/deploy.go

```diff
@@ -44,12 +46,19 @@ func newDeployCommand(dockerCli command.Cli) *cobra.Command {
 	addRegistryAuthFlag(&opts.sendRegistryAuth, flags)
 	flags.BoolVar(&opts.prune, "prune", false, "Prune services that are no longer referenced")
 	flags.SetAnnotation("prune", "version", []string{"1.27"})
+	flags.BoolVar(&opts.noResolveImage, "no-resolve-image", false, "Do not query the registry to resolve image digest and supported platforms")
+	flags.SetAnnotation("no-resolve-image", "version", []string{"1.30"})
 	return cmd
 }
 
 func runDeploy(dockerCli command.Cli, opts deployOptions) error {
 	ctx := context.Background()
 
+	// image resolution should not happen for clients older than v1.30
```

Comment:
Even though this says "ClientVersion()" it's important to note it actually reflects the supported version of the server. The comment might be a bit misleading in that sense, but I guess it's fine.